app/model/RuleContext.scala

```diff
@@ -17,61 +17,73 @@
 package model
 
 import connector._
-import play.api.mvc.{AnyContent, Request}
+import javax.inject.{Inject, Singleton}
 import play.api.{Logger, LoggerLike}
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
-
-case class RuleContext(credId: Option[String])(implicit request: Request[AnyContent], hc: HeaderCarrier, ec: ExecutionContext) {
-
-  val request_ = request
-  val hc_ = hc
+@Singleton
+class RuleContext @Inject()(val authConnector: AuthConnector,
+                            selfAssessmentConnector: SelfAssessmentConnector)(implicit ec: ExecutionContext)
+  extends AuthorisedFunctions {
 
   val logger: LoggerLike = Logger
 
-  val selfAssessmentConnector: SelfAssessmentConnector = SelfAssessmentConnector
-  val authConnector: FrontendAuthConnector = FrontendAuthConnector
-  val userDetailsConnector: UserDetailsConnector = UserDetailsConnector
-
-  lazy val userDetails = authority.flatMap { authority =>
-    authority.userDetailsUri.map(userDetailsConnector.getUserDetails).getOrElse {
-      logger.warn("failed to get user details because userDetailsUri is not defined")
-      Future.failed(new RuntimeException("userDetailsUri is not defined"))
-    }
+  def isVerifyUser(implicit hc: HeaderCarrier): Future[Boolean] = {
+    authorised(AuthProviders(AuthProvider.Verify)){
```

Comment:
Will this force a login if they are not authorised? or does it merely check to see what the situation is?